pages/_app.js

```diff
@@ -1,10 +1,36 @@
 import GlobalStyle from "../styles";
+import { useState } from "react";
+import { initialShoppingList } from "@/lib/data";
+import { uid } from "uid";
+import sortShoppingListByCategory from "@/utils/sortShoppingListByCategory";
 
 export default function App({ Component, pageProps }) {
+  const [showForm, setShowForm] = useState(false);
+  const [currentTasks, setCurrentTasks] = useState(initialShoppingList);
```

Comment:
I recommend renaming this state to:

```suggestion
  const [shoppingItems, setShoppingItems] = useState(initialShoppingList);
```

This will make it a lot clearer what is being stored in the state.

The consequence will be that you will need to adjust the naming in some locations, but it is worth the effort